pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1854,120 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "single % character", input: `%`, validInput: false},
+		{description: "escaped % character", input: `%%`, validInput: true},
+		{description: "escaped % and only a % character", input: `%%%`, validInput: false},
+		{description: "two literal % characters", input: `%%%%`, validInput: true},
+		{description: "zero percent", input: `%%%%%%0`, validInput: true},
+		{description: "escaped expression", input: `%%[XYZ.hdr(Host)]\ %[XYZ.hdr(Host)]`, validInput: true},
+		{description: "simple empty expression", input: `%[]`, validInput: false},
+		{description: "nested empty expressions", input: `%[%[]]`, validInput: false},
+		{description: "empty quoted value", input: `%{+Q}`, validInput: false},
+		{description: "quoted value", input: `%{+Q}foo`, validInput: false},
+		{description: "quoted value", input: `%{+Q} ssl_c_der`, validInput: false},
+		{description: "hdr with empty field", input: `%[XYZ.hdr()]`, validInput: false},
+		{description: "hdr with percent field", input: `%[XYZ.hdr(%)]`, validInput: false},
+		{description: "hdr with known field", input: `%[XYZ.hdr(Host)]`, validInput: true},
+		{description: "hdr with syntax error", input: `%[XYZ.hdr(Host]`, validInput: false},
+
+		{description: "hdr with valid X-XSS-Protection value", input: `1;mode=block`, validInput: true},
+		{description: "hdr with valid Content-Type value", input: `text/plain,text/html`, validInput: true},
+
+		{description: "incomplete expression", input: `%[req`, validInput: false},
+		{description: "quoted field", input: `%[XYZ.hdr(%{+Q}Host)]`, validInput: false},
+		{description: "value with conditional expression", input: `%[XYZ.hdr(Host)] if foo`, validInput: true},
+		{description: "value with what looks like a conditional expression", input: `%[XYZ.hdr(Host)]\ if\ foo`, validInput: true},
+		{description: "unsupported fetcher and converter", input: `%[date(3600),http_date]`, validInput: false},
+		{description: "not allowed sample fetchers", input: `%[foo,lower]`, validInput: false},
+		{description: "not allowed converters", input: `%[req.hdr(host),foo]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{Q[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{{Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host)`, validInput: false},
+		{description: "missing parentheses or braces", input: `%req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdrhost)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{req.hdr(host)}`, validInput: false},
```

Comment:
nit: each of these tests should have a unique description.

Reply:
ok added.

Reply:
https://github.com/openshift/library-go/commit/398bba91d32aa80169a697d8b06ba0ff5edc5774#diff-51cbcb9e560e512d7dba0cd93dab386055814a27e2c7903ad93bce92c58fd18bR1903